File: typescript/src/bubble-sort.ts
function bubbleSort(arr: Array<number>) {
	let size = arr.length - 1

	while(size > 1) {
		for (let i in arr) {
			const index = Number(i)

			if (index < arr.length - 1) {
				const current = arr[index]
				const next = arr[index + 1]

				if (next < current) {
					arr[index] = next
					arr[index + 1] = current
				}
			}
		}

		size -= 1
	}

	return arr
}

const myArr = [87,3,28,7,23,3,25,6,328,21]
console.log(bubbleSort(myArr))

File: python/src/bubble_sort.py
from typing import List

def bubble_sort(numbers: List[int]):
	size = len(numbers) - 1

	while size > 1:
		for index, item in enumerate(numbers):
			if index < len(numbers) - 1:
				current = numbers[index]
				next = numbers[index + 1]

				if next < current:
					numbers[index] = next
					numbers[index + 1] = current

		size -= 1

	return numbers

if __name__ == "__main__":
	numbers = [7,13,73,6,68,71,23,4,37,4,36]
	print(
		bubble_sort(numbers)
	)